includes/Core/Util/Beta_Migration.php

```diff
@@ -111,7 +111,8 @@ function () {
 	public function migrate_old_credentials() {
 		$credentials = $this->credentials->get();
 
-		if ( $credentials['oauth2_client_id'] && ! strpos( $credentials['oauth2_client_id'], '.apps.sitekit.withgoogle.com' ) ) {
+		// Credentials can be filtered in so we must also check if there is a saved option present.
+		if ( $this->options->get( Credentials::OPTION ) && $credentials['oauth2_client_id'] && ! strpos( $credentials['oauth2_client_id'], '.apps.sitekit.withgoogle.com' ) ) {
```

Comment:
Let's only rely on the option here and not on the `Credentials` class. Someone may theoretically have the option set to a non-proxy client ID, but filter in a proxy client ID. Unlikely, but I think it's safer to only check the option because that's what this is about. We'll need to add `! empty()` around the checks, but that's fine.

Reply:
@felixarntz I thought about that but then we'd need to get `Data_Encryption` class involved since the value is encrypted. If you're OK with adding that in here, then we can do that 👍

Reply:
We could use `Encrypted_Options`. I think adding that here would make sense.